### ISSUE TITLE
Remote modules can no longer statically import data URLs

### ISSUE THREAD

**Issue by 0f-0b** [denoland/deno#17914]
The command mentioned in the [1.7 release notes](https://deno.com/blog/v1.7#support-for-importing-data-urls) now reports an error.

```shellsession
$ deno run https://deno.com/blog/v1.7/import_data_url.ts
error: Remote modules are not allowed to import local modules. Consider using a dynamic import instead.
  Importing: data:application/typescript;base64,ZXhwb3J0IGNvbnN0IGEgPSAiYSI7CgpleHBvcnQgZW51bSBBIHsKICBBLAogIEIsCiAgQywKfQo=
    at https://deno.com/blog/v1.7/import_data_url.ts:2:20
$ deno --version
deno 1.31.0 (release, aarch64-apple-darwin)
v8 11.0.226.13
typescript 4.9.4
```

**Comment by dsherret** [comment]
This one is only a regression when using an import map. It was actually broken in past versions as well without one:

```
> deno run https://deno.com/blog/v1.7/import_data_url.ts
error: Remote modules are not allowed to import local modules. Consider using a dynamic import instead.
  Importing: data:application/typescript;base64,ZXhwb3J0IGNvbnN0IGEgPSAiYSI7CgpleHBvcnQgZW51bSBBIHsKICBBLAogIEIsCiAgQywKfQo=
    at https://deno.com/blog/v1.7/import_data_url.ts:2:20
V:\scratch
> deno -V
deno 1.30.3
```

**Comment by dsherret** [comment]
Should be fixed in 1.31.1
